vyper/compiler/output.py

```diff
@@ -206,7 +207,15 @@ def build_ir_runtime_dict_output(compiler_data: CompilerData) -> dict:
 
 
 def build_metadata_output(compiler_data: CompilerData) -> dict:
-    sigs = compiler_data.function_signatures
+    _ = compiler_data.function_signatures
+    module_t = compiler_data.annotated_vyper_module._metadata["type"]
+    sigs = dict[str, ContractFunctionT]()
+
+    for fn_t in module_t.exposed_functions:
+        assert isinstance(fn_t.ast_def, vy_ast.FunctionDef)
+        for rif_t in fn_t.reachable_internal_functions:
+            sigs[rif_t.ast_def.module_node.path + ": " + rif_t.name] = rif_t
```

Comment:
i think there can actually be collisions between `path` since it's not resolved (e.g. depending on which search path is used)